compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java

```diff
@@ -142,6 +142,11 @@ public Location location() {
             return Optional.of(this.location);
         }
 
+        if (this.tSymbol != null) {
+            this.location = this.tSymbol.pos;
+            return Optional.of(this.location);
+        }
```

Comment:
This seems like an arbitrary hack. Dig a bit deeper to identify what the issue is, why it is occurring and at what place can it be fixed.

Reply:
When the `tSymbol != tSymbol.type.tsymbol`, the location has to be either set in the constructor or obtained from the `tSymbol.pos` (within the `getLocation()` method). 
Updated to explicitly set the location in the constructor.